File: cpp/08_002.cpp
#include <vector>
#include <unordered_map>

std::vector<int> findPricePair(std::vector<int>& prices, 
                              int budget) {
    std::unordered_map<int, int> priceMap;
    
    for (int i = 0; i < prices.size(); i++) {
        int complement = budget - prices[i];
        
        if (priceMap.find(complement) != priceMap.end()) {
            return {priceMap[complement], i};
        }
        
        priceMap[prices[i]] = i;
    }
    
    return {};  // No solution found
}

File: java/08_002.java
import java.util.HashMap;
import java.util.Map;

public int[] findPricePair(int[] prices, int budget) {
    Map<Integer, Integer> priceMap = new HashMap<>();
    
    for (int i = 0; i < prices.length; i++) {
        int complement = budget - prices[i];
        
        if (priceMap.containsKey(complement)) {
            return new int[] {priceMap.get(complement), i};
        }
        
        priceMap.put(prices[i], i);
    }
    
    return new int[] {};  // No solution found
}